source/server/admin/logs_handler.cc

```diff
@@ -56,56 +61,106 @@ bool LogsHandler::changeLogLevel(const Http::Utility::QueryParams& params) {
     return false;
   }
 
-  std::string name = params.begin()->first;
-  std::string level = params.begin()->second;
+  const auto& it = params.begin();
+  absl::string_view key{it.first};
+  absl::string_view value{it.second};
+
+  if (key == "all") {
+    // Change all log levels.
+    return changeAllLogLevels(value);
+  } else if (key == "paths") {
+    // Bulk change log level by name:level pairs, separated by comma.
+
+    // Build a map of name:level pairs, a few allocations is ok here since it's
+    // not common to call this function at a high rate.
+    absl::flat_hash_map<absl::string_view, absl::string_view> name_levels;
+    std::vector<absl::string_view> pairs = absl::StrSplit(value, ',', absl::SkipWhitespace());
+    for (const auto& name_value : pairs) {
+      std::pair<absl::string_view, absl::string_view> name_value_pair =
+          absl::StrSplit(name_value, absl::MaxSplits(':', 1), absl::SkipWhitespace());
+      auto [name, value] = name_value_pair;
+      if (name.empty() || value.empty()) {
+        continue;
+      }
+
+      name_levels.insert({name, value});
```

Comment:
nit: `name_levels[name] = value` feels slightly more readable to me.

Another difference is that with what you have, the first one wins when there's a duplicate name.. With what I suggest, the last one wins which also seems slightly more intuitive.

Another option is to use insert() like you have it but check the return value and error out on a duplicate.